### PR TITLE
Restore httpEnableTimelineLogging state after hot restart

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This partially addresses https://github.com/flutter/devtools/issues/1968. To completely fix the issue, we need a socket rpc change like https://github.com/dart-lang/sdk/commit/e85e5b6264edf554c13d328a0af9c38672c4a0c3 @bkonyi

### CODE REVIEW

File: packages/devtools_app/lib/src/service_manager.dart [hunk 1]
```diff
@@ -587,42 +587,52 @@ class ServiceExtensionManager {
         break;
       case 'Flutter.ServiceExtensionStateChanged':
         final String name = event.json['extensionData']['extension'].toString();
-        final String valueFromJson =
+        final String valueString =
```
Comment (jacob314) [inline packages/devtools_app/lib/src/service_manager.dart:590]: nit: just call this value not valueString. No need to include types in variable names.
Also consider removing the type of the LHS as the type is already inferred by having the toString.

File: packages/devtools_app/lib/src/service_manager.dart [hunk 2]
```diff
@@ -587,42 +587,52 @@ class ServiceExtensionManager {
...
             event.json['extensionData']['value'].toString();
+        await _updateServiceExtensionForStateChange(name, valueString);
+        break;
+      case 'HttpTimelineLoggingStateChange':
+        final name = extensions.httpEnableTimelineLogging.extension;
+        final valueString = event.json['extensionData']['enabled'].toString();
+        await _updateServiceExtensionForStateChange(name, valueString);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/service_manager.dart:597]: here and elsewhere. valueString -> value

File: packages/devtools_app/lib/src/service_manager.dart [hunk 3]
```diff
@@ -587,42 +587,52 @@ class ServiceExtensionManager {
...
+        name,
+        enabled,
+        value,
+        callExtension: false,
+      );
     }
   }
 
-  dynamic _getExtensionValueFromJson(String name, String valueFromJson) {
+  dynamic _getExtensionValue(String name, String valueString) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/service_manager.dart:625]: maybe valueString -> encodedValue here and elsewhere if `value` is confusing.

File: packages/devtools_app/lib/src/ui/service_extension_widgets.dart [hunk 4]
```diff
@@ -421,17 +421,19 @@ mixin _ServiceExtensionMixin<T extends _ServiceExtensionWidget> on State<T> {
...
       await action();
 
-      if (widget.completedText != null) {
+      if (mounted && widget.completedText != null) {
         Notifications.of(context).push(widget.completedText);
       }
     } catch (e, st) {
       log('$e\n$st');
 
       Notifications.of(context).push(widget.describeError(e));
```
Comment (jacob314) [inline packages/devtools_app/lib/src/ui/service_extension_widgets.dart:430]: shouldn't we also short circuit displaying the notification if the widget is not mounted?

**Review by jacob314 — COMMENTED** [review]
Can we test this?

File: packages/devtools_app/lib/src/service_manager.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/service_manager.dart]: done

File: packages/devtools_app/lib/src/service_manager.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/service_manager.dart]: done

File: packages/devtools_app/lib/src/service_manager.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/service_manager.dart]: done

File: packages/devtools_app/lib/src/ui/service_extension_widgets.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/ui/service_extension_widgets.dart]: done

**Comment by kenzieschmoll** [comment]
the [hot restart tests](https://github.com/flutter/devtools/blob/709fa720442af8d219dee391729c5bc4b6ea41a8/packages/devtools_testing/lib/service_manager_test.dart#L207) are currently commented out due to https://github.com/flutter/devtools/issues/337. I could take a look at seeing if they can be re-enabled.

**Comment by jacob314** [comment]
lgtm

**Comment by kenzieschmoll** [comment]
Landing this with a TODO to address the hot restart tests

**Pull request merged**